### ISSUE TITLE
Launched containers get status "(suspended)"

### ISSUE THREAD

**Issue by electrical** [jenkinsci/docker-plugin#119]
Jenkins: 1.590
Plugin: 0.8
Docker: 1.3.1

When launching a job it often get in the status "suspended" and waits a long time before starting up a new instance which may or not fall in the same status.
Log shows nothing special.

Any idea's ?

**Comment by wsoula** [comment]
I think I am having the same issue.  If I set the instance cap to 1, I get a slave spun up in the (suspended) state and the build never builds.  If I set the instance cap to 2, I get two slaves in (suspended) state.  If I set it to 3, the log shows two of them being retried twice before I have two slaves that can build things.

**Comment by oier** [comment]
I have the same issue too the containers remains as (suspended) even when the build was sucessfull and the container was removed from the docker host machine

**Comment by mgarratt** [comment]
I am getting this issue as well, the slaves are running SSH and a few other things with Supervisor and appear to be launching and running properly. Eventually one will work, but it can take a long time.

**Comment by skord** [comment]
Same problem here. I can disconnect and reconnect the agent and the agent says it's fine, but the nodes are suspended. I don't see anything wrong with the slaves other than being in a suspended state. Builds are dead in the water. 

* Jenkins: 1.598
* Docker 1.3.2
* Plugin: 0.8

**Comment by samrocketman** [comment]
Seems that a timeout needs to be extended (or configurable) for when it asks for a docker agent and when it actually gets one.  It seems to hit a timeout ("oops haven't gotten it yet so I'll try to provision again") and it continues that cycle creating many slaves.

**Comment by gpmidi** [comment]
@samrocketman That would make sense given the behavior we're seeing.

**Comment by KostyaSha** [comment]
For me Jenkins deletes nodes after timeout, have you tried to wait?

**Comment by KostyaSha** [comment]
Btw, instead of deleting jenkins slaves, try run `docker ps -a` and then `docker rm` all stopped instances.

**Comment by andrechrist** [comment]
Same behavior for me, Build is successful but slaves stay in suspended, offline mode. It seems, as if there is a connection attempt AFTER the build when the docker container is already gone. Is there any issue with the retention strategy used in the plugin???

* Jenkins: 1.598
* Docker 1.5.0
* Plugin: 0.8

Logfile shows SSH connection issue:
```
[02/14/15 21:24:09] [SSH] Opening SSH connection to localhost:49164.
Connection refused
SSH Connection failed with IOException: "Connection refused".
java.io.IOException: There was a problem while connecting to localhost:49164
	at com.trilead.ssh2.Connection.connect(Connection.java:818)
	at com.trilead.ssh2.Connection.connect(Connection.java:687)
	at com.trilead.ssh2.Connection.connect(Connection.java:587)
	at hudson.plugins.sshslaves.SSHLauncher.openConnection(SSHLauncher.java:1146)
	at hudson.plugins.sshslaves.SSHLauncher$2.call(SSHLauncher.java:701)
	at hudson.plugins.sshslaves.SSHLauncher$2.call(SSHLauncher.java:696)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.net.ConnectException: Connection refused
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:339)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:200)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:182)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:579)
	at com.trilead.ssh2.transport.TransportManager.establishConnection(TransportManager.java:354)
	at com.trilead.ssh2.transport.TransportManager.initialize(TransportManager.java:467)
	at com.trilead.ssh2.Connection.connect(Connection.java:758)
	... 9 more
[02/14/15 21:24:09] Launch failed - cleaning up connection
[02/14/15 21:24:09] [SSH] Connection closed.
```

**Comment by samrocketman** [comment]
Using the script I described it deletes the containers when it deletes the slaves.  So using docker rm would just be an added step.  The ssh connectivity issue is what I witnessed but that is because it didn't wait long enough for the container to come up.  This is something that should be configurable.

**Comment by andrechrist** [comment]
@samrocketman Ok, understand. Where do you put your script?

**Comment by samrocketman** [comment]
So the script, [mentioned earlier][s], can be run one of two ways.

1. Execute it in the Jenkins script console (_Manage Jenkins > Script Console_)
2. Create a new freestyle project and add an Execute Groovy Script step.  Paste the script in that step and use it as a job.  This would require you to install the groovy plugin.

[s]: https://github.com/jenkinsci/docker-plugin/issues/119#issuecomment-66233107

**Comment by BitOne** [comment]
It seams that this PR could help on the delay problem between launch of the container and opening of the SSH connection: https://github.com/jenkinsci/docker-plugin/pull/150

**Comment by KostyaSha** [comment]
Should be resolved with 0.10.0. 
Please reopen if issue will be not resolved.
